src/main/scala/com/ibm/couchdb/api/builders/QueryStrategy.scala

```diff
@@ -16,39 +16,57 @@
 
 package com.ibm.couchdb.api.builders
 
-import com.ibm.couchdb.Req
+import com.ibm.couchdb.api.Query
 import com.ibm.couchdb.core.Client
-import org.http4s.Status
+import com.ibm.couchdb.{CouchDocs, CouchView, Res, TypeMapping}
 import upickle.default.Aliases.{R, W}
 import upickle.default._
 
+import scala.reflect.ClassTag
 import scalaz.concurrent.Task
 
-trait QueryStrategy {
+case class QueryBasic[C: R](
+    client: Client, db: String,
+    params: Map[String, String] = Map.empty,
+    ids: Seq[String] = Seq.empty) {
+  def query: Task[C] = {
+    val url = s"/$db/_all_docs"
+    ids match {
+      case Nil => QueryUtils.query[C](client, db, url, params)
+      case _ => QueryUtils.queryByIds[String, C](client, db, url, ids, params)
+    }
+  }
+}
 
-  def postQuery[Q: R](
-      client: Client,
-      db: String,
-      url: String,
-      params: Map[String, String]): Task[Q] = {
-    client.get[Q](url, Status.Ok, params.toSeq)
+case class QueryIncludeDocs[C: R, D: R](
```

Comment:
This class seems to be exactly the same as QueryBasic, except or the type parameter D.